docs/source/contributing/developing.adoc

```diff
@@ -225,23 +225,42 @@ To run all the test, use the following command:
 $ make integration_all
 ----
 
-Flag *_TIMEOUT_* can be used to override the default timeout `3600s`.
-To run all the test with timeout `7200s`, use the following command:
+===== Additional Parameters
+
+To provide more flexibility the default targets `integration` and `integration_all` can be further customized using several parameters.
+
+EXEC_BEFORE_FEATURE::
+Parameter `EXEC_BEFORE_FEATURE` specifies Minishift commands to be run before each feature.
+This provides ability to run integration tests against Minishift which is not in default state.
+When multiple commands are specified, they must be delimited by a semicolon.
+For example, tests can be run against stopped Minishift with _image caching_ option turned on by running:
 
 ----
-$ make integration_all TIMEOUT=7200s
+$ make integration_all EXEC_BEFORE_FEATURE="start; stop; config set image-caching true"
```

Comment:
+1, though it looks heavy :)

Reply:
@agajdosi I am little confused. If someone wants to set image-caching to true Why would we need to start and stop the minishift instance to enable image-caching?

Reply:
@agajdosi Also what do you think about `BEFORE_FEATURE_RUN` instead of `EXEC_BEFORE_FEATURE`. I understand that `EXEC` stands for execute but `RUN` is more easy to understand IMO.